epix/io.py

```diff
@@ -80,75 +81,106 @@ def loader(directory,
             eventids, instead of rows.
 
     Returns:
         awkward1.records: Interactions (eventids, parameters, types).
         integer: Number of events simulated.
     """
-    ttree, n_simulated_events = _get_ttree(directory, file_name)
+    file = os.path.join(directory, file_name)
 
-    if arg_debug:
-        print(f'Total entries in input file = {ttree.num_entries}')
-        cutby_string='output file entry'
-        if cut_by_eventid:
-            cutby_string='g4 eventid'
-
-        if kwargs_uproot_arrays['entry_start'] is not None:
-            print(f'Starting to read from {cutby_string} {kwargs_uproot_arrays["entry_start"]}')
-        if kwargs_uproot_arrays['entry_stop'] is not None:
-            print(f'Ending read in at {cutby_string} {kwargs_uproot_arrays["entry_stop"]}')
-
-    # If user specified entry start/stop we have to update number of
-    # events for source rate computation:
-    if kwargs_uproot_arrays['entry_start'] is not None:
-        start = kwargs_uproot_arrays['entry_start']
-    else:
-        start = 0
-
-    if kwargs_uproot_arrays['entry_stop'] is not None:
-        stop = kwargs_uproot_arrays['entry_stop']
-    else:
-        stop = n_simulated_events
-    n_simulated_events = stop - start
-
-    if cut_by_eventid:
-        # Start/stop refers to eventid so drop start drop from kwargs
-        # dict if specified, otherwise we cut again on rows.
-        kwargs_uproot_arrays.pop('entry_start', None)
-        kwargs_uproot_arrays.pop('entry_stop', None)
-
-    # Columns to be read from the root_file:
-    column_names = ["x", "y", "z", "t", "ed",
-                    "type", "trackid",
-                    "parenttype", "parentid",
-                    "creaproc", "edproc"]
-
-    # Conversions and parameters to be computed:
-    alias = {'x': 'xp/10',  # converting "geant4" mm to "straxen" cm
-             'y': 'yp/10',
-             'z': 'zp/10',
-             'r': 'sqrt(x**2 + y**2)',
-             't': 'time*10**9'
-             }
+    if not os.path.isfile(file):
+        print("File does not exist!")
+        #cancel epix execution here
 
+    #Prepare cut for root and csv case
     if outer_cylinder:
         cut_string = (f'(r < {outer_cylinder["max_r"]})'
-                      f' & ((zp >= {outer_cylinder["min_z"] * 10}) & (zp < {outer_cylinder["max_z"] * 10}))')
+                    f' & ((zp >= {outer_cylinder["min_z"] * 10}) & (zp < {outer_cylinder["max_z"] * 10}))')
     else:
         cut_string = None
 
-    # Read in data, convert mm to cm and perform a first cut if specified:
-    interactions = ttree.arrays(column_names,
-                                cut_string,
-                                aliases=alias,
-                                **kwargs_uproot_arrays)
-    eventids = ttree.arrays('eventid', **kwargs_uproot_arrays)
-    eventids = ak.broadcast_arrays(eventids['eventid'], interactions['x'])[0]
-    interactions['evtid'] = eventids
+
+    if file.endswith(".root"):
```

Comment:
This if statement indicates that the subsequent part should be an additional function. If the user wants to read whats going on for root files he/she can jump to the corresponding function. If not one can directly continue to the next code.